src/utils.ts

```diff
@@ -171,7 +171,10 @@ export function applyRetryableWrites<T extends HasRetryableWrites>(target: T, db
  * @param value - An object that could be a promise
  * @returns true if the provided value is a Promise
  */
-export function isPromiseLike<T = any>(value?: PromiseLike<T> | void): value is Promise<T> {
+export function isPromiseLike<T = any>(
+  value?: PromiseLike<T> | void | unknown
+): value is Promise<T> {
+  // @ts-expect-error since value exists, value.then check is valid ts
   return !!value && typeof value.then === 'function';
 }
```

Comment:
```suggestion
export function isPromiseLike<T = any>(value?: unknown): value is PromiseLike<T> {
  return (
    value != null &&
    typeof value === 'object' &&
    'then' in value &&
    typeof value.then === 'function'
  );
}
```

We can just use `unknown` as the input, which forces us to prove a few more facts about the value but all worth doing since we do not know what we may be passing in.

Reply:
Sounds good, I've changed it!